src/MSBuildLocator/MSBuildLocator.cs

```diff
@@ -209,6 +209,27 @@ public static void RegisterMSBuildPath(string[] msbuildSearchPaths)
             // AssemblyResolve event can fire multiple times for the same assembly, so keep track of what's already been loaded.
             var loadedAssemblies = new Dictionary<string, Assembly>();
 
+            // MSBuild can be loaded from the x86 or x64 folder. Before 17.0, it looked next to the executing assembly in some cases and constructed a path that assumed x86 in others.
+            // This overrides the latter assumption to let it find the right MSBuild.
+            foreach (string path in msbuildSearchPaths)
```

Comment:
Do we not need to do the `amd64` folder removal dance here too, in case someone passed in a 64-bit folder?

Can we do this only for <17.0?